face/UnifiedFaceCollection/python/unified_face_collection.py

```diff
@@ -0,0 +1,346 @@
+        persons_response = requests.get(persons_url, headers=self.headers)
+        persons = persons_response.json()
+        for person in persons:
+            if person['name'] == person_name:
+                return person
+        return None
+
+    def train(self):
+        # Train the Large Face List
+        face_list_train_url = f"{self.face_api_url}/largefacelists/{self.large_face_list_id}/train"
+        face_list_train_response = requests.post(face_list_train_url, headers=self.headers)
+        
+        if face_list_train_response.status_code != 202:
+            print("Failed to initiate training for Large Face List")
+            return False
+
+        # Train the Large Person Group
+        person_group_train_url = f"{self.face_api_url}/largepersongroups/{self.large_person_group_id}/train"
+        person_group_train_response = requests.post(person_group_train_url, headers=self.headers)
+        
+        if person_group_train_response.status_code != 202:
+            print("Failed to initiate training for Large Person Group")
+            return False
+
+        # Check training status for Large Face List
+        face_list_status_url = f"{self.face_api_url}/largefacelists/{self.large_face_list_id}/training"
+        while True:
+            face_list_status_response = requests.get(face_list_status_url, headers=self.headers)
+            face_list_status = face_list_status_response.json()
+            if face_list_status['status'] == 'succeeded':
+                break
+            elif face_list_status['status'] == 'failed':
+                print("Large Face List training failed")
+                return False
+            time.sleep(1)
+        
+        # Check training status for Large Person Group
+        person_group_status_url = f"{self.face_api_url}/largepersongroups/{self.large_person_group_id}/training"
+        while True:
+            person_group_status_response = requests.get(person_group_status_url, headers=self.headers)
+            person_group_status = person_group_status_response.json()
+            if person_group_status['status'] == 'succeeded':
+                break
+            elif person_group_status['status'] == 'failed':
+                print("Large Person Group training failed")
+                return False
+            time.sleep(1)
+        
+        return True
+
+    def find_face(self, image_path, search_type='face'):
+        # Detect face in the image
+        detected_faces = detect_faces(self.subscription_key, self.endpoint, image_path, self.injection_header)
+        face_ids = [face['faceId'] for face in detected_faces]
+        if not face_ids:
+            return []
+
+        normalized_results = []
+        if search_type == 'person':
+            # Find similar persons in the Large Person Group
+            identify_url = f"{self.face_api_url}/identify"
+            identify_data = {
+                'faceId': face_ids[0],
```

Comment:
Looks like an undocumented parameter? I only see "faceId**s**" instead of "faceId" according to the API docs: https://learn.microsoft.com/rest/api/face/face-recognition-operations/identify-from-large-person-group?#request-body.

Reply:
My bad. Fixed. Thanks for catching.